dlib/media/ffmpeg_utils.h

```diff
@@ -382,6 +410,23 @@ namespace dlib
     namespace ffmpeg
     {
 
+// ---------------------------------------------------------------------------------------------------
+
+        namespace details
+        {
+            template<class... Args>
+            inline bool fail(std::ostream& out, Args&&... args)
+            {
+#ifdef __cpp_fold_expressions
+                ((out << args),...);
+#else
+                (void)std::initializer_list<int>{((out << args), 0)...};
+#endif
+                out << '\n';
+                return false;
```

Comment:
Why return the bool?  It's always `false`.

Reply:
It's so you can write something like:
```
if (!something_that_can_fail())
    return fail("reason for failing");
```

Reply:
Yeah I guess I've come around to this :)